src/bootstrap/exec.rs

```diff
@@ -0,0 +1,48 @@
+use std::process::Command;
+
+/// What should be done when the command fails.
+#[derive(Debug, Copy, Clone)]
+pub enum BehaviorOnFailure {
+    /// Immediately stop bootstrap.
+    Exit,
+    /// Delay failure until the end of bootstrap invocation.
+    DelayFail,
+}
+
+/// How should the output of the command be handled.
+#[derive(Debug, Copy, Clone)]
+pub enum OutputMode {
+    /// Print both the output (by inheriting stdout/stderr) and also the command itself, if it
+    /// fails.
+    PrintAll,
+    /// Print the output (by inheriting stdout/stderr).
+    PrintOutput,
+    /// Suppress the output if the command succeeds, otherwise print the output.
+    Suppress,
+}
+
+/// Wrapper around `std::process::Command`.
+#[derive(Debug)]
+pub struct BootstrapCommand<'a> {
+    pub command: &'a mut Command,
+    pub failure_behavior: Option<BehaviorOnFailure>,
```

Comment:
All `BehaviorOnFailure` does is either exit on failure, or keep the execution until the end. In this case `fail_fast: bool` should be able to replace `Option<BehaviorOnFailure>` here ?

Reply:
Good point. I didn't realize that when this is `None`, the error will be "swallowed", which we don't want to do and which wasn't done by bootstrap before. I would still keep the enum to make the behavior more explicit, just removing the option and defaulting to `BehaviorOnFailure::Exit`. What do you think?